src/umockdev-record.vala

```diff
@@ -417,7 +428,41 @@ main (string[] args)
     foreach (string s in opt_evemu_events)
         record_script(s, "evemu");
 
-    Posix.execvp(opt_devices[0], opt_devices);
-    exit_error("Cannot run program %s: %s", opt_devices[0], strerror(errno));
-    return 0;
+    int status;
+    try {
+        Process.spawn_async (null, opt_devices, null,
+                            SpawnFlags.SEARCH_PATH | SpawnFlags.CHILD_INHERITS_STDIN | SpawnFlags.DO_NOT_REAP_CHILD ,
+                            null, out child_pid);
+    } catch (Error e) {
+            stderr.printf ("Cannot run %s: %s\n", opt_devices[0], e.message);
+            Process.exit (1);
+    }
+
+    // propagate signals to the child
+    var act = Posix.sigaction_t() { sa_handler = child_sig_handler, sa_flags = Posix.SA_RESETHAND };
```

Comment:
This whole commit is basically a copy of how umockdev-run spawns its child. As this is far from trivial, let's try to factorize this.

Reply:
Yeah. Ideally, I would love for all this to live inside `umockdev.vala`. i.e. make recording into a special operation mode of the testbed.

I have not figured out how to do that at this point though (didn't put too much thought into that so far though). So, maybe best to just have a small set of utilities elsewhere.

Reply:
OK, moved it all into a `umockdev-utils.vala` file, which gets compiled into a static library.

There is probably a bit more cleanup potential there now.